src/backend/auxil/range_alloc.rs

```diff
@@ -49,14 +55,19 @@ where
                 }
             });
         }
-        best_fit.map(|(index, range)| {
-            if range.end - range.start == length {
-                self.free_ranges.remove(index);
-            } else {
-                self.free_ranges[index].start += length;
+        match best_fit {
+            Some((index, range)) => {
+                if range.end - range.start == length {
+                    self.free_ranges.remove(index);
+                } else {
+                    self.free_ranges[index].start += length;
+                }
+                Ok(range.start..(range.start + length))
             }
-            range.start..(range.start + length)
-        })
+            None => Err(RangeAllocationError {
+                fragmented_free_length,
```

Comment:
Rather than using a `match` here you could just call `ok_or` on the output from the original map function.